### DISCUSSION TITLE
X.md wanted as X.html, not X\index.html

### DISCUSSION THREAD

**Discussion by ronzulu** [squidfunk/mkdocs-material#7406] [Q&A] [answered]
Hi

Definitely a newbie question, and not sure if this is the right place to ask.

I'm using mkdocs to generate a static site that I want to host on S3.

The issue seems to be that mkdocs creates a folder for every source markdown file, and converting the markdown to a file called index.html, regardless of the source filename.

This means that S3 needs to serve `x/index.html` when `x` is requested. Although that can be easily configured for the root level, it doesn't seem so easy to do that anywhere else.

Is there some way to configure mkdocs to simply generate `x.html` instead of its default behaviour?

I only have some web development experience. I'm curious about the rationale for mkdocs creating a folder for every source markdown file. Is that a convention used by static web sites in general?

Many thanks

Ronny

**Comment by squidfunk** [comment, marked as answer]
Add this to your `mkdocs.yml`, [as explained in MkDocs' documentation](https://www.mkdocs.org/user-guide/configuration/#use_directory_urls):

``` yaml
use_directory_urls: false
```

**Comment by ronzulu** [comment]
Thanks for the super quick response!

My bad... I did see that option, but the description `controls the style used for linking to pages` made me think that it didn't control the structure of the generated output. Just something stylistic.

Cheers
Ronny

**Reply by squidfunk** [reply]
We don't control the MkDocs documentation, so maybe you could raise this suggestion to [MkDocs](https://github.com/mkdocs/mkdocs) itself? ☺️

**Comment by ronzulu** [comment]
Thanks for the suggestion @squidfunk I've suggested this in https://github.com/mkdocs/mkdocs/issues/3797
